heartbeat/ocf-distro

```diff
@@ -5,43 +5,205 @@
 # released RH specific stuff (/etc/sysconfig/nfs editing)
 # 
 
+# systemd-based systems should all have an os-release file.
+_ETC_OS_RELEASE_FILE="/etc/os-release"
+_USR_OS_RELEASE_FILE="/usr/lib/os-release"
+
+# Legacy distro-specific files
 _DEBIAN_VERSION_FILE="/etc/debian_version"
 _REDHAT_RELEASE_FILE="/etc/redhat-release"
 _SUSE_RELEASE_FILE="/etc/SuSE-release"
-_RELEASE_FILES="/etc/*-release"
-_REDHAT_BASED_DISTROS_RE='red *hat|fedora|centos|scientific linux'
 
-get_release_id() {
+_DEBIAN_BASED_DISTROS_RE="debian|ubuntu"
+_REDHAT_BASED_DISTROS_RE="red *hat|rhel|fedora|centos|scientific"
+_SUSE_BASED_DISTROS_RE="sles|suse"
+
+
+# Convert OS release ID to a standard form regardless of source.
+_process_os_release_id() {
+	_os="$1"
+
+	# Isolate distro name, remove whitespace, convert to lowercase
+	_os=$(echo "$_os" | sed -e "s/\([G][Nn][Uu]\/\)\?[Ll][Ii][Nn][Uu][Xx]//" \
+		-e "s/[Ss][Ee][Rr][Vv][Ee][Rr]//" \
+		-e "s/release.*//" -e "s/[[:digit:]].*//" \
+		| tr -d '[:blank:]' | tr '[:upper:]' '[:lower:]')
+
+	# Normalize known distros to os-release names
+	case "$_os" in
+	*centos*)
+		_os="centos"
+		;;
+	*debian*)
+		_os="debian"
+		;;
+	*fedora*)
+		_os="fedora"
+		;;
+	*redhat*|*rhel*|*scientific*)
+		_os="rhel"
+		;;
+	*opensuse*)
+		_os="opensuse"
+		;;
+	*suseenterprise*)
+		_os="sles"
+		;;
+	*ubuntu*)
+		_os="ubuntu"
+		;;
+	esac
+
+	echo "$_os"
+}
+
+# Convert OS version ID to a form that ocf_version_cmp() can handle.
+# Strip any garbage.
+_process_os_version_id() {
+	_ver="$1"
+
+	# Discard everything in _ver except the version
+	_pattern="[[:digit:]]\([[:digit:].-]*[[:digit:]]\)\?"
+	_ver=$(echo "$_ver" | sed "s/\(^\|.*[[:blank:]]\)\($_pattern\).*/\2/")
+
+	# Append ".0" for integers so that ocf_version_cmp doesn't fail
+	if [ "$_ver" -eq "$_ver" ] 2>/dev/null; then
```

Comment:
`[ "$_ver" -eq "$_ver" ]` is only true if `$_ver` is an integer. This is a simple and AFAIK portable idiom to test whether `$_ver` is an integer and then append `.0` if it is. We also need error redirection in case `$_ver` is not an integer. See below.
```
# _ver="1"
# [ "$_ver" -eq "$_ver" ] && echo true || echo false
true

# _ver="1.0"
# [ "$_ver" -eq "$_ver" ] && echo true || echo false
-bash: [: 1.0: integer expression expected
false
```

Reply:
Ah. I thought it was a typo :+1: .

Reply:
The updated version adds the `".0"` using sed.